keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,649 @@
+   extended resource.
+
+With these additions in place, the scheduler has everything it needs to support
+the dynamic allocation of devices to requests made through extended resource.
+kubelet and DRA driver has everything it needs to admit and pass the allocated
+devices to the pod to run.
+
+## Design Details
+
+### Resource Slice API
+The exact set of proposed API changes on Resource Slice can be seen below:
+```go
+// ResourceSliceSpec contains the information published by the driver in one ResourceSlice.
+type ResourceSliceSpec struct {
+	...
+
+	// The extended resource name for all the devices in the ResourceSlice
+	// advertised as
+	//
+	// +optional
+	ExtendedResourceName string
+}
+
+// Device represents one individual hardware instance that can be selected based
+// on its attributes. Besides the name, exactly one field must be set.
+// +k8s:deepcopy-gen=true
+type Device struct {
+	// Name is unique identifier among all devices managed by
+	// the driver in the pool. It must be a DNS label.
+	//
+	// +required
+	Name string `json:"name"`
+	...
+
+	// ExtendedResourceName is the extended resource name
+	// the device is advertised as. It must be a DNS label.
+	//
+	// +optional
+	ExtendedResourceName string
+}
+```
+
+The devices can be advertised with an extended resource name. The extended
+resource name can be specified on each individual device. Different
+devices can be advertised as different extended resource name, or not
+advertised as extended resource at all.
+
+Alternatively, the extended resource name can be specified at the
+`ResourceSlice` level, then all the devices in the resource slice are
+advertised as the given extended resource name. If a device has a different
+extended resource name than that given in the `ResoureSlice`, the device's
+extended resource name is used for that device.
+
+### Resource Claim API
+There is no API change on Resource Claim. However, a special resource claim is
+created to keep track of device allocations for extended resource. The special
+resource claim has following properties:
+
+  * It is namespace scoped, like other resource claims. Unlike resource slice,
+    which is cluster scoped.
+  * It is a singleton. There is at most one resource claim object for a given
+    extended resource in a given namespace.
+  * It is not owned by a pod, its owner reference is null.
+  * Its field `status.allocation.devices` is used, other fields are ununsed,
+    such as `status.allocation.reservedFor` and `spec`.
+  * It has annotation `resource.kubernetes.io/extended-resource-name:`, and it
+    does not have annotation `resource.kubernetes.io/pod-claim-name:`
+
+```yaml
+metadata:
+  annotations:
+    resource.kubernetes.io/extended-resource-name: foo.domain/bar
+```
+
+The special resource claim lifecycle is managed by the resource claim
+controller.
+
+  * It is *created* in a namespace when there is a pod with extended resource
```

Comment:
And I assume it gets created in the same namespace as the pod?

Reply:
Yes, it is in the same namespace. clarified it in the KEP.